src/commands/mod.rs

```diff
@@ -780,6 +840,229 @@ impl DaemonControl {
             .map_err(CommandError::TxBroadcast)
     }
 
+    /// Create PSBT to replace the given transaction using RBF.
+    ///
+    /// If the previous transaction includes a change output to one of our own change addresses,
+    /// this same address will be used for change in the RBF transaction, if required. If the previous
+    /// transaction pays to more than one of our change addresses, then the one receiving the highest
+    /// value will be used as a change address and the others will be treated as non-change outputs.
+    ///
+    /// `is_cancel` indicates whether to "cancel" the transaction by removing non-change outputs from
+    /// the replacement or otherwise to keep the same non-change outputs and simply bump the fee.
+    /// If `true`, the only output of the RBF transaction will be change and the inputs will include
+    /// at least one of the inputs from the previous transaction. If `false`, all inputs from the previous
+    /// transaction will be used in the replacement.
+    /// In both cases, the RBF transaction may include additional confirmed coins as inputs if required
+    /// in order to pay the higher fee (this applies also when replacing a self-send).
+    ///
+    /// `feerate_vb` is the target feerate for the RBF transaction (in sat/vb). If `None`, it will be set
+    /// to 1 sat/vb larger than the feerate of the previous transaction, which is the minimum value allowed
+    /// when using RBF.
+    pub fn rbf_psbt(
+        &self,
+        txid: &bitcoin::Txid,
+        is_cancel: bool,
+        feerate_vb: Option<u64>,
+    ) -> Result<CreateSpendResult, CommandError> {
+        // The transaction to be replaced must be one of ours and so we can assume it signals replaceability.
+        // The inputs in the replacement transaction will be either those coins used as inputs in the transaction
+        // to be replaced or "confirmed" coins. Therefore, we can assume the transaction to be replaced is
+        // the only one directly conflicting with the replacement transaction.
+        let mut db_conn = self.db.connection();
+
+        let prev_psbt = db_conn
+            .spend_tx(txid)
+            .ok_or(CommandError::UnknownSpend(*txid))?;
+        let prev_coins = {
+            let prev_outpoints: &Vec<bitcoin::OutPoint> = &prev_psbt
+                .unsigned_tx
+                .input
+                .iter()
+                .map(|txin| txin.previous_output)
+                .collect();
+            let prev_coins = db_conn.coins_by_outpoints(prev_outpoints);
+            assert_eq!(prev_outpoints.len(), prev_coins.len());
```

Comment:
Presumably the tx would have been dropped because its fees were below the mempool minfee so they would need to bump the fees before re-broadcasting it.

Reply:
I now check for outpoints for which we have no coin in the DB and return an error.